registration/api/v2/serializers.py

```diff
@@ -164,12 +165,47 @@ class Meta:
         }
 
 
+# pylint: disable=abstract-method
+class ThemeSchemaSerializerGenerator(serializers.Serializer):
+    """
+    Custom automatically generated serializer from the theme schemas.
+
+    This is needed to make the Swagger code generator correctly generate the
+    schema model in the frontend without requiring to duplicate the schema
+    there.
+    """
+    def __init__(self, *args, **kwargs):
+        """
+        Start with empty serializer and add fields from both theme schemas
+        """
+        super().__init__(*args, **kwargs)
+
+        theme_schema_combined = {
+            **theme_schema_v0['properties'],
+            **theme_schema_v1['properties']
+        }
+        for key, value in theme_schema_combined.items():
+            field_type = None
+            if key == 'version':
+                field_type = serializers.IntegerField()
+            elif value == ref('flag'):
+                field_type = serializers.BooleanField(required=False)
+            else:
+                field_type = serializers.CharField(
+                    max_length=7,
+                    required=False
+                    # TODO: Add a color validator here
```

Comment:
All attributes except version and flag will always be colors? I suggest that we only assume that the attribute is a color if its name ends with "color" and only validate it in this case.

Reply:
>I noticed that currently all attributes except version and flag are colors, but couldn't that change in the future? I'd prefer to avoid that assumption that all these attributes are colors.

Yes, that's a good point, but I have no of what the fields will look like in the future. I can change the `else` statement to `elif value == ref('color')` if that makes the solution more acceptable.

Reply:
> I can change the else statement to elif value == ref('color') if that makes the solution more acceptable.

+1

Reply:
>Yes, that's the code duplicating part of the data definition done in the theme_schema_v1 JSON Schema in instance/schemas/theming.py. I'm still getting familiar with the code and right now cannot point out a better solution, but this duplication does not look good. I'd at least add a TODO tag.

This serializer actually avoids duplication schema definition code and allows the frontend API client to be automatically generated without needing to add serializer fields one by one.